Added recipe to have an interactive shell when using docker

Thanush-03 opened the pull request:
Issue number
Fixes #1960
Description:
Added documentation for interactive shell debugging when using Docker containers. This provides the Docker equivalent of the bin/instance debug command used in traditional Plone installations.
The new "Interactive shell for debugging" section in the Docker recipes documentation includes:
Multiple approaches for getting an interactive shell:
Using docker exec for running containers
Using docker run for new container instances
Using docker compose exec for Docker Compose setups
Practical examples showing how to access the Plone site and set up fake requests
Safety warnings about using this in production environments.

LINK:
https://6.docs.plone.org/install/containers/recipes/index.html

MESSAGE:
The issue being fixed #1960
 Interactive shell debugging documentation was added in Docker recipes page.


<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1967.org.readthedocs.build/

<!-- readthedocs-preview plone6 end -->

Thanush-03 committed:
Added recipe to have an interactive shell when using docker

Thanush-03 committed:
Fix CI submodule issues and add interactive shell documentation

- Remove duplicate documentation directory that was causing submodule errors
- Add submodules: true to CI workflows to properly handle submodules
- Add comprehensive interactive shell debugging documentation for Docker containers

Fixes #1960

stevepiercy commented:
@Thanush-03 please see [Granting permission to publish](https://6.docs.plone.org/contributing/documentation/index.html#granting-permission-to-publish). We can't accept your contribution without your explicit written permission. Thank you!

Thanush-03 commented:
Thankyou sir for review the work what I done.


I,Thanush K, agree to have this contribution published under Creative
Commons 4.0 International License (CC BY 4.0), with attribution to the
Plone Foundation.
Thank you sir.

On Sat, 16 Aug 2025, 10:18 pm Steve Piercy, ***@***.***>
wrote:

> *stevepiercy* left a comment (plone/documentation#1967)
> <https://github.com/plone/documentation/pull/1967#issuecomment-3193782669>
>
> @Thanush-03 <https://github.com/Thanush-03> please see Granting
> permission to publish
> <https://6.docs.plone.org/contributing/documentation/index.html#granting-permission-to-publish>.
> We can't accept your contribution without your explicit written permission.
> Thank you!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/plone/documentation/pull/1967#issuecomment-3193782669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BR3YYI4A2NHPJJXSDVQDEGL3N5OHNAVCNFSM6AAAAACEBTSID6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCOJTG44DENRWHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

stevepiercy commented:
@Thanush-03 is "Thanush K" your actual full name? I've never seen a surname consisting of a single letter, except for Malcom X.

Also did you test your instructions to make sure that they actually work? If you haven't actually tested the instructions, then please close this PR. Thank you!

Thanush-03 commented:
Yes my full name Thanush K ,sorry for that confusion.I have tested the
instructions it was worked.

On Sat, 16 Aug 2025, 10:30 pm Steve Piercy, ***@***.***>
wrote:

> *stevepiercy* left a comment (plone/documentation#1967)
> <https://github.com/plone/documentation/pull/1967#issuecomment-3193787511>
>
> @Thanush-03 <https://github.com/Thanush-03> is "Thanush K" your actual
> full name? I've never seen a surname consisting of a single letter, except
> for Malcom X.
>
> Also did you test your instructions to make sure that they actually work?
> If you haven't actually tested the instructions, then please close this PR.
> Thank you!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/plone/documentation/pull/1967#issuecomment-3193787511>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BR3YYI24RYVIQ4XDBYCVBRT3N5PSRAVCNFSM6AAAAACEBTSID6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCOJTG44DONJRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Thanush-03 committed:
Fix deprecated plone.volto:default-homepage profile references

- Remove references to deprecated plone.volto:default-homepage profile
- Update documentation to recommend content export/import approach
- Add explanatory notes about the profile removal in version 5
- Update Docker example to remove deprecated profile usage

Fixes #118

Thanush-03 committed:
Merge branch '6.0' into thanush/1960

Thanush-03 commented:
@stevepiercy  kindly review this and provide me any changes required

stevepiercy commented on .github/workflows/build_deploy.yml:
```diff
@@ -16,6 +16,8 @@ jobs:
       url: https://docs.plone.org
     steps:
       - uses: actions/checkout@v5
+        with:
+          submodules: true
```
Why did you add this? It's not broken and it has nothing to do with the purpose of this PR. Please revert.

```suggestion
```

stevepiercy commented on .github/workflows/test.yml:
```diff
@@ -7,6 +7,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v5
+        with:
+          submodules: true
```
Why did you add this? It's not broken and it has nothing to do with the purpose of this PR. Please revert.

```suggestion
```

stevepiercy commented:
@davisagli, @mauritsvanrees, @ericof, @sneridagh would you please do a technical review of the changes to the documentation regarding `plone.volto` and the container recipe? Thank you!

stevepiercy commented on docs/developer-guide/create-a-distribution.md:
````diff
@@ -331,6 +337,11 @@ bin/export-distribution path/to/zope.conf Plone
 
 In the example above, `Plone` is the ID of the Plone site to export.
 
+```{note}
+This is the recommended approach for creating example content in distributions.
+The old approach of using profiles like `plone.volto:default-homepage` has been deprecated and removed from `plone.volto` in version 5.
````
@davisagli can I get a confirmation about this change?

````suggestion
```{versionremoved} `plone.volto` 5.0.0b1
This is the recommended approach for creating example content in distributions.
The old approach of using profiles like `plone.volto:default-homepage` has been deprecated and removed from [`plone.volto` in version 5.0.0b1](https://github.com/plone/plone.volto/blob/main/CHANGES.md#500b1-2024-10-30).
````

davisagli replied:
@stevepiercy It is correct that the `plone.volto:default-homepage` profile was removed from plone.volto, but I think it's noise to mention that here. These docs are for someone creating a new distribution, not someone who is updating an existing distribution that might have referenced that profile.

It's incorrect to say that the old approach of referencing content profiles in the distribution is deprecated. Both approaches for creating example content are still supported. It's fine to say that this one is the recommended one (since it requires less configuration).

I don't understand why any of this is in this particular pull request, which claims to be about adding documentation for an interactive shell, which is unrelated to distributions.

davisagli replied:
I made a [separate PR](https://github.com/plone/documentation/pull/1969) to clean up the outdated references to the plone.volto:default-homepage profile, so that we can focus on the interactive shell docs here.

stevepiercy commented on docs/install/containers/images/frontend.md:
```diff
@@ -108,7 +108,8 @@ services:
       - SITE=Plone
       - 'ADDONS=plone.restapi==8.21.0 plone.volto==4.0.0a3 plone.rest==2.0.0a2 plone.app.iterate==4.0.2 plone.app.vocabularies==4.3.0'
-      - 'PROFILES=plone.volto:default-homepage'
+      # Note: plone.volto:default-homepage profile was removed in version 5
+      # Use content export/import instead for example content
```
Well, wait a minute. The `ADDONS` includes `plone.volto==4.0.0a3`, and not `5.0.0b1` or later, making this comment very confusing. @davisagli @ericof @sneridagh @mauritsvanrees can I get some clarification on this?

davisagli replied:
This is a quite old list of version pins which was probably left over from Plone 5 where volto wasn't included. Neither the ADDONS nor PROFILES environment variables should be needed here; the plone-backend image already contains appropriate versions.

stevepiercy commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+When developing or troubleshooting Plone applications in Docker containers, you may need an interactive Python shell with the Plone environment loaded.
+This is the Docker equivalent of the `bin/instance debug` command used in traditional Plone installations.
+
+### Using docker exec
```
```suggestion
### `docker exec`
```

stevepiercy commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+
+### Using docker exec
+
+If you have a running Plone container, you can start an interactive shell using `docker exec`:
```
```suggestion
In a running Plone container, start an interactive shell using `docker exec` by substituting the name of your backend container for `BACKEND_CONTAINER_NAME` in the following command.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+docker exec -it <container_name> /bin/bash
+```
+
+Once inside the container, you can start the Plone debug console:
````
```suggestion
Once inside the container, start the Plone debug console.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+bin/instance debug
+```
+
+### Using docker run
````
```suggestion
### `docker run`
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+bin/instance debug
+```
+
+### Using Docker Compose
````
```suggestion
### `docker compose`
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+docker run -it --rm plone/plone-backend:{PLONE_BACKEND_MINOR_VERSION} /bin/bash
+```
+
+Then start the debug console:
````
```suggestion
Then start the debug console.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+
+### Using Docker Compose
+
+If you're using Docker Compose, you can run an interactive shell with:
```
```suggestion
With `docker compose`, run an interactive shell with the following command.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+docker compose exec backend /bin/bash
+```
+
+Then start the debug console:
````
```suggestion
Then start the debug console.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+bin/instance debug
+```
+
+Or run it directly in one command:
````
```suggestion
Or run it directly in one command.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+docker compose exec backend bin/instance debug
+```
+
+### Accessing the Plone site
````
```suggestion
### Access the Plone site
```

stevepiercy commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+
+### Accessing the Plone site
+
+Once in the debug console, you can access your Plone site and perform debugging operations:
```
```suggestion
Once in the debug console, you can access your Plone site and perform debugging operations.
```

davisagli replied:
This information isn't specific to a console that was started in a container. It should go under https://6.docs.plone.org/admin-guide/run-plone.html#run-a-debug-console instead.

stevepiercy commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+# Access the root application
+app = self.app
+
+# Access your Plone site (replace 'Plone' with your site ID)
+site = app['Plone']
+
+# Access content
+folder = site['my-folder']
+item = folder['my-item']
```
Assume code uses Black formatting.

```suggestion
# Access your Plone site (replace "Plone" with your site ID)
site = app["Plone"]

# Access content
folder = site["my-folder"]
item = folder["my-item"]
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+setRequest(app.REQUEST)
+```
+
+### Exiting the debug console
````
```suggestion
### Exit the debug console
```

stevepiercy commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+
+### Using docker run
+
+Alternatively, you can start a new container instance specifically for debugging:
```
```suggestion
Alternatively, start a new container instance specifically for debugging.
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+If you have a running Plone container, you can start an interactive shell using `docker exec`:
+
+```shell
+docker exec -it <container_name> /bin/bash
````
docker exec -it <container_name> /bin/bash

```suggestion
docker exec -it BACKEND_CONTAINER_NAME /bin/bash
```

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+Once inside the container, you can start the Plone debug console:
+
+```shell
+bin/instance debug
````
I think there's a step missing before this one. I followed the instructions in https://6.docs.plone.org/install/containers/index.html through "Access your Plone site" to set up the containers. The `bin` directory has all the usual Plone stuff, but no `instance`.

```shell
docker exec -it plone6-backend /bin/bash
root@9ae16356b1a2:/app# bin/instance debug
bash: bin/instance: No such file or directory
```

davisagli replied:
Within the plone-backend container, the correct command is `./docker-entrypoint.sh console`, not `bin/instance debug`. `bin/instance debug` would only work with a buildout-based installation, which the plone-backend container does not use.

stevepiercy commented on docs/install/containers/recipes/index.md:
````diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+Then start the debug console:
+
+```shell
+bin/instance debug
````
See previous comment. I have no `bin/instance`.

stevepiercy reviewed (changes requested):
I only made a few style changes.

This looks really good, with the exception that I couldn't find my `bin/instance` so couldn't run a debugger. What am I missing?

davisagli commented on docs/install/containers/recipes/index.md:
```diff
@@ -170,3 +170,89 @@ docker compose run backend pack
+## Interactive shell for debugging
+
+When developing or troubleshooting Plone applications in Docker containers, you may need an interactive Python shell with the Plone environment loaded.
+This is the Docker equivalent of the `bin/instance debug` command used in traditional Plone installations.
```
There are multiple traditions for how to install Plone. It'd be clearer to say buildout-based instead of traditional.

Note, we already have documentation for doing this outside the context of a Docker container in https://6.docs.plone.org/admin-guide/run-plone.html#run-a-debug-console where we call it a debug console rather than an interactive shell. I don't care what we call it but we should be consistent.

stevepiercy commented:
@Thanush-03 would you please rebase your branch on `main`, resolve any merge conflicts, and update your pull request according to the reviews here? Then we can do another review and proceed. Thank you!

stevepiercy commented:
@Thanush-03 are you still working on this PR? Please acknowledge.
